src/ir/possible-contents.cpp

```diff
@@ -1275,6 +1275,31 @@ struct InfoCollector
   void visitContNew(ContNew* curr) {
     // TODO: optimize when possible
     addRoot(curr);
+
+    // The function reference that is passed in here will be called, just as if
+    // we were a call_ref, except at a potentially later time.
+    if (!curr->func->type.isRef()) {
+      return;
+    }
+    auto targetType = curr->func->type.getHeapType();
+    if (!targetType.isSignature()) {
+      assert(targetType.isBottom());
```

Comment:
This should never happen, unless I'm missing something special about GUFA?

Reply:
This is similar to the logic for call_indirect/ref:

https://github.com/WebAssembly/binaryen/blob/920ea767cf03620345691bd7a372d4f14b44806b/src/ir/possible-contents.cpp#L815-L821

It handles the case of `(cont.new (ref.null))`. Given we've ruled out unreachable earlier, all that is left is either a proper Signature type, or a null/bottom.